File: src/main/java/com/kamikazejam/datastore/base/store/CacheLoggerInstantiator.java
package com.kamikazejam.datastore.base.store;

import com.kamikazejam.datastore.base.Cache;
import com.kamikazejam.datastore.base.log.LoggerService;
import org.jetbrains.annotations.NotNull;

public interface CacheLoggerInstantiator {

    @NotNull
    LoggerService instantiate(Cache<?,?> cache);

}

File: src/main/kotlin/com/kamikazejam/datastore/base/store/CacheLoggerInstantiator.kt
package com.kamikazejam.datastore.base.store

import com.kamikazejam.datastore.base.Cache
import com.kamikazejam.datastore.base.log.LoggerService

fun interface CacheLoggerInstantiator {
    fun instantiate(cache: Cache<*, *>): LoggerService
}
